scripts/coccinelle/unsigned_assignment.cocci

```diff
@@ -0,0 +1,40 @@
+// Find assignments to unsigned variables and add an 'U' to the value
+// Copyright (C) 2018 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+virtual patch
+virtual report
+
+@r_unsigned@
+typedef uint8_t, uint16_t, uint32_t, uint64_t;
+{unsigned char, unsigned short, unsigned int, uint8_t, uint16_t, uint32_t, uint64_t} v;
+constant C;
+position p;
+@@
+
+  v = C@p
+
+@script:python r_rewrite@
+C << r_unsigned.C;
+z;
+@@
+
+if C.isdigit() != True:
+    cocci.include_match(False)
```

Comment:
Now get rid of these two lines above.

Reply:
...removing the isdigit() check down here causes the patch mode to add an extra 'U' in ext/ like so:
`diff -u -p a-project/ext/hal/cmsis/Include/mpu_armv8.h b-project/ext/hal/cmsis/Include/mpu_armv8.h
--- a-project/ext/hal/cmsis/Include/mpu_armv8.h
+++ b-project/ext/hal/cmsis/Include/mpu_armv8.h
@@ -270,7 +270,7 @@ __STATIC_INLINE void ARM_MPU_SetRegion_N
 __STATIC_INLINE void orderedCpy(volatile uint32_t* dst, const uint32_t* __RESTRICT src, uint32_t len)
 {
   uint32_t i;
-  for (i = 0U; i < len; ++i) 
+  for (i = 0UU; i < len; ++i) 
   {
     dst[i] = src[i];
   }
`
I have yet to figure out whether this too is caused by version 1.0.4.

Reply:
No, it isn't caused by 1.0.4!

```
himanshu@himanshu-Vostro-3559:~/zephyr$ cat test.cocci test.c; spatch --sp-file test.cocci test.c
@r@
{unsigned short, unsigned int} v;
constant C;
@@

* v = C

int main (void)
{
	unsigned int foo = 20U;
}
init_defs_builtins: /usr/local/bin/../lib/coccinelle/standard.h
HANDLING: test.c
diff = 
--- test.c
+++ /tmp/cocci-output-5327-3dff8c-test.c
@@ -1,5 +1,4 @@
 
 int main (void)
 {
-	unsigned int foo = 20U;
 }
```

Instead `constant` matches `0U` in this case and adds an extra `U`.
So, isdigit() hack is good to go unless we have any methods to
check for an existing `U`.